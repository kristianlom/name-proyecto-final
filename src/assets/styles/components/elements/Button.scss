@import '../../lib/fonts';
@import  '../../lib/palette';

.Button {
  width: 283px;
  height: 50px;
  background: map-get(map-get($palette,lightTheme),white) 0% 0% no-repeat padding-box;
  box-shadow: 0px 0px 4px #00000029;
  border-radius: 25px;
  opacity: 1;
  font-size: 14px;
  font-family: 'Quicksand', sans-serif;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 5px 0;
  &:focus {
    outline: none;
  }
  &--success {
    border: 2px solid map-get(map-get($palette,alert),success);
    color: map-get(map-get($palette,greyColors),black);
  }
  &--error {
    border: 2px solid map-get(map-get($palette,alert),failure);
    color: map-get(map-get($palette,greyColors),black);
  }
  &--blue {
    background-color: map-get(map-get($palette,buttons),blue);
    color: map-get(map-get($palette,lightTheme),white);
  }
  &--gradient {
    color: map-get(map-get($palette,lightTheme),white);
    background: transparent linear-gradient(90deg, #5533FF 0%, #0BD6FC 100%) 0% 0% no-repeat padding-box;
  }
}