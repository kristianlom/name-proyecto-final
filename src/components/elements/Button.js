/* eslint-disable react/button-has-type */
import React from 'react';
import '../../assets/styles/components/elements/Button.scss';
import GoogleText from './GoogleText';

const Button = (props) => {
  return (
    <>
      {
        props.name === 'blueBase' && (
          <button type={props.buttonType} className="Button Button--base-blue">
            {props.children}
          </button>
        )
      }
      {
        props.name === 'purpleBase' && (
          <button type={props.buttonType} className="Button Button--base-purple">
            {props.children}
          </button>
        )
      }
      {
        props.name === 'google' && (
          <button type={props.buttonType} className="Button Button--image">
            {props.children}
            <GoogleText />
          </button>
        )
      }
      {
        props.name === 'success' && (
          <button type={props.buttonType} className="Button Button--success">
            {props.children}
          </button>
        )
      }
      {
        props.name === 'error' && (
          <button type={props.buttonType} className="Button Button--error">
            {props.children}
          </button>
        )
      }
      {
        props.name === 'blue' && (
          <button type={props.buttonType} className="Button Button--blue">
            {props.children}
          </button>
        )
      }
      {
        props.name === 'gradient' && (
          <button type={props.buttonType} className="Button Button--gradient">
            {props.children}
          </button>
        )
      }
    </>
  );
};

export default Button;
