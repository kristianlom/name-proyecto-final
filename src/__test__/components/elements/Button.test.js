import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../../components/elements/Button';

const wrapped = shallow(<Button />);

describe('snapshot of Button component', () => {
  it('should render the Button Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });
  it('renders the Titles children', () => {
    expect(wrapped.find(Button).length).toEqual(1);
  });
});

