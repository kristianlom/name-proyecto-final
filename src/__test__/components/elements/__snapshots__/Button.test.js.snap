// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of Button component should render the Button Component correctly 1`] = `ShallowWrapper {}`;
